cmd/util/flags/cliflags/job.go

```diff
@@ -0,0 +1,232 @@
+package cliflags
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"github.com/bacalhau-project/bacalhau/cmd/util/flags"
+	"github.com/bacalhau-project/bacalhau/cmd/util/parse"
+	"github.com/bacalhau-project/bacalhau/pkg/model"
+	"github.com/bacalhau-project/bacalhau/pkg/models"
+)
+
+type JobSettings struct {
+	name        string
+	namespace   string
+	jobType     string
+	priority    int
+	count       int
+	constraints string
+	labels      map[string]string
+
+	// TODO(forrest): remove these fields and their usage when we complete deprecation of legacy flag names.
+	// tracked via https://github.com/bacalhau-project/bacalhau/issues/3838
+	legacy *LegacyJobFlags
+	// we hold a reference to the command so we may check if users provided legacy or new flag names.
+	cmd *cobra.Command
+}
+
+func (j *JobSettings) Name() string {
+	return j.name
+}
+
+func (j *JobSettings) Namespace() string {
+	return j.namespace
+}
+
+func (j *JobSettings) Type() string {
+	if j.cmd.Flags().Changed("target") {
+		jobType := models.JobTypeBatch
+		if j.legacy.targetingMode == model.TargetAll {
+			jobType = models.JobTypeOps
+		}
+		return jobType
+	}
+	return j.jobType
```

Comment:
Lets discuss with product about this flag and moving from `--target` to `--job-type`. As I mentioned before, I am questioning enabling users to submit long running jobs using declarative approach